### PR TITLE
[CTB-124] Wallet Sell endpoint

### PR DESCRIPTION
**Pull request by mik-sondej**

### CODE REVIEW

File: src/server/constants.py [hunk 1]
```diff
@@ -67,3 +67,5 @@ class QUERIES:
                         LIMIT 1"""
     WALLET_BUY_ADD_BTC: Query = "UPDATE users SET wallet_btc = wallet_btc + %s WHERE uuid=%s"
     WALLET_BUY_SUBTRACT_USD: Query = "UPDATE users SET wallet_usd = wallet_usd - %s WHERE uuid=%s"
+    WALLET_SELL_SUBTRACT_BTC: Query = "UPDATE users SET wallet_btc = wallet_btc - %s WHERE uuid=%s"
+    WALLET_SELL_ADD_USD: Query = "UPDATE users SET wallet_usd = wallet_usd + %s WHERE uuid=%s"
```
Comment (PiotrDeda) [inline src/server/constants.py:71]: `WALLET_BUY_SUBTRACT_USD` and `WALLET_SELL_ADD_USD` are the same as `WALLET_WITHDRAW` and `WALLET_DEPOSIT` respectively. Maybe it's better to use those instead?

Comment (unlvy) [reply src/server/constants.py:71]: This should be dicussed with @vis4rd

File: src/server/wallet/wallet_service.py [hunk 2]
```diff
@@ -115,3 +115,39 @@ def buy(uuid: str, amount: float) -> Response:
...
+
+        """
+        with DatabaseProvider.handler() as handler:
+            handler().execute(QUERIES.SELECT_USER_DATA_BY_UUID, (uuid,))
+            user_data: tuple[str, str, str] = handler().fetchone()
+            if user_data:
+                # Check if user has enough BTC to perform transaction
+                if float(user_data[2]) >= amount:
+                    handler().execute(QUERIES.SELECT_LATEST_STOCK_PRICE)
+                    price: tuple[str,] = handler().fetchone()
```
Comment (vis4rd) [inline src/server/wallet/wallet_service.py:138]:
```suggestion
                    price: tuple[str] = handler().fetchone()
```

File: src/server/responses.py [hunk 3]
```diff
@@ -131,6 +139,14 @@ def not_enough_money_to_make_a_purchase() -> Response:
             409,
         )
 
+    @staticmethod
+    def not_enough_BTC_to_make_a_sale() -> Response:
+        """409: user tried to sell more BTC that they own."""
```
Comment (vis4rd) [inline src/server/responses.py:144]:
```suggestion
        """409: user tried to sell more BTC than they own."""
```

File: src/server/constants.py (hunk 1 above)
Comment (vis4rd) [reply src/server/constants.py:71]: For sure buy and sell operations should be separate from withdraw and deposit, even though for database it's virtually the same query. I think that for now let's keep it separate for consistency, we can worry about duplication once it hits us with some bug.

Comment (PiotrDeda) [inline src/server/constants.py:71]: For better support of transaction history, these two queries should be merged – see #39 (make sure to update test code too)

**Review by PiotrDeda — CHANGES_REQUESTED** [review]

**Review by vis4rd — APPROVED** [review]

**Pull request merged**